### PR TITLE
feat: detach fromRepository with force import

### PR DESCRIPTION
**Pull request by yesnault**

### CODE REVIEW

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#15035.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Docs -   
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔ 
* Package and Analyze
  * Docker ✔ 
  * Sonar ✔ 
* Integration
  * Docker Compose Tests ✘  
  * Standalone Tests ✘

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#15036.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Docs -   
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#15037.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Docs -   
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#15037.1 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Docs -   
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#15039.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Docs -   
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

**Review by richardlt — APPROVED** [review]

**Review by fsamin — APPROVED** [review]

File: engine/api/ascode/dao.go [hunk 1]
```diff
@@ -95,3 +95,45 @@ func deleteEvent(db gorp.SqlExecutor, event *sdk.AsCodeEvent) error {
...
+	AND (data -> 'pipelines' ->> $2)::text = $3
+	AND (data -> 'workflows')::text = 'null'
+	AND (data -> 'environments')::text = 'null'
+	AND (data -> 'applications')::text = 'null'
+	`
+	_, err := db.Exec(query, fromRepository, pipID, pipName)
+	return sdk.WithStack(err)
+}
+
+func DeleteEventsApplicationOnlyFromPipelineName(ctx context.Context, db gorp.SqlExecutor, fromRepository string, appID int64, appName string) error {
```
Comment (sguiheux) [inline engine/api/ascode/dao.go]: PipelineName ?

File: engine/api/ascode/dao.go [hunk 2]
```diff
@@ -95,3 +95,45 @@ func deleteEvent(db gorp.SqlExecutor, event *sdk.AsCodeEvent) error {
...
+	AND (data -> 'applications' ->> $2)::text = $3
+	AND (data -> 'workflows')::text = 'null'
+	AND (data -> 'environments')::text = 'null'
+	AND (data -> 'pipelines')::text = 'null'
+	`
+	_, err := db.Exec(query, fromRepository, appID, appName)
+	return sdk.WithStack(err)
+}
+
+func DeleteEventsEnvironmentOnlyFromPipelineName(ctx context.Context, db gorp.SqlExecutor, fromRepository string, envID int64, envName string) error {
```
Comment (sguiheux) [inline engine/api/ascode/dao.go]: EnvironmentName

**Review by sguiheux — CHANGES_REQUESTED** [review]

**Review by sguiheux — APPROVED** [review]

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#15043.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Docs -   
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

**Pull request merged**